### ISSUE TITLE
unable to open output file 'compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o': 'Operation not permitted'

### ISSUE THREAD

**Issue by AA1999** [llvm/llvm-project#57465]
Hi, I get this error while building llvm:

```console
[40/43] Performing build step for 'runtimes'
[9/738] Building CXX object compiler-rt/lib/ubsan/CMak...bsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o
FAILED: compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o
/home/arshia/Desktop/llvm-project/build/./bin/clang++ --target=x86_64-unknown-linux-gnu -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/arshia/Desktop/llvm-project/compiler-rt/lib/ubsan/.. -O3 -pipe -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Wall -Wno-unused-parameter -g -m32 -fPIC -fno-builtin -fno-exceptions -fomit-frame-pointer -funwind-tables -fno-stack-protector -fno-sanitize=safe-stack -fvisibility=hidden -fno-lto -Wthread-safety -Wthread-safety-reference -Wthread-safety-beta -O3 -gline-tables-only -Wno-gnu -Wno-variadic-macros -Wno-c99-extensions -nostdinc++ -fno-rtti -DUBSAN_CAN_USE_CXXABI -Wno-format -std=c++17 -MD -MT compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o -MF compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o.d -o compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o -c /home/arshia/Desktop/llvm-project/build/runtimes/runtimes-bins/compiler-rt/lib/ubsan/dummy.cpp
error: unable to open output file 'compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.i386.dir/dummy.cpp.o': 'Operation not permitted'
1 error generated.
[10/738] Building CXX object compiler-rt/lib/ubsan/CMa...an_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o
FAILED: compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o
/home/arshia/Desktop/llvm-project/build/./bin/clang++ --target=x86_64-unknown-linux-gnu -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/arshia/Desktop/llvm-project/compiler-rt/lib/ubsan/.. -O3 -pipe -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Wall -Wno-unused-parameter -g -m64 -fPIC -fno-builtin -fno-exceptions -fomit-frame-pointer -funwind-tables -fno-stack-protector -fno-sanitize=safe-stack -fvisibility=hidden -fno-lto -Wthread-safety -Wthread-safety-reference -Wthread-safety-beta -O3 -gline-tables-only -Wno-gnu -Wno-variadic-macros -Wno-c99-extensions -nostdinc++ -fno-rtti -DUBSAN_CAN_USE_CXXABI -Wno-format -std=c++17 -MD -MT compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o -MF compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o.d -o compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o -c /home/arshia/Desktop/llvm-project/build/runtimes/runtimes-bins/compiler-rt/lib/ubsan/dummy.cpp
error: unable to open output file 'compiler-rt/lib/ubsan/CMakeFiles/RTUbsan_dynamic_version_script_dummy.x86_64.dir/dummy.cpp.o': 'Operation not permitted'
1 error generated.
[16/738] Building CXX object compiler-rt/lib/asan/CMakeFiles/RTAsan_dynamic.i386.dir/asan_interceptors.cpp.o
ninja: build stopped: subcommand failed.
FAILED: runtimes/runtimes-stamps/runtimes-build /home/arshia/Desktop/llvm-project/build/runtimes/runtimes-stamps/runtimes-build
cd /home/arshia/Desktop/llvm-project/build/runtimes/runtimes-bins && /usr/bin/cmake --build .
ninja: build stopped: subcommand failed.
```

My config is:
```shell
cmake -S llvm -B build -G Ninja -DLLVM_ENABLE_PROJECTS='clang;clang-tools-extra;cross-project-tests;flang;libc;libclc;lld;lldb;openmp;polly' -DLLVM_ENABLE_RUNTIMES='libcxx;libcxxabi;compiler-rt;libunwind' -DLLVM_BUILD_DOCS=OFF -DLLVM_ENABLE_BINDINGS=OFF -DLIBCXX_CXX_ABI=libcxxabi
```

**Comment by AA1999** [comment]
Bumping because it seems buried.

**Comment by AA1999** [comment]
Fixed with #60770
